### DISCUSSION TITLE
👨‍💻☎ WinUI Community Call: February 15, 2023

### DISCUSSION THREAD

**Discussion by gabbybilka** [microsoft/microsoft-ui-xaml#8137] [Announcements]
### https://www.youtube.com/watch?v=xcgc5CLqUkM

# Details
Date: February 15, 2023
Time: 17:00-18:00 UTC (9:00-10:00am Pacific)

Anyone and everyone is welcome - no pre-registration is required.
This will be an informal interactive live stream directly with members of our engineering team.

# Format
The community call is a call among the WinUI team that is live-streamed onto YouTube. We present on new updates, share information, welcome guests, and answer your questions.

### **Agenda**
- Intro
- Spotlight on new features introduced in 1.3: 
  - @DmitriyKomin: Xaml Backdrop APIs
  - @pratikone: Window.AppWindow and latest updates around custom title bars 
- Latest news regarding WinAppSDK, WinUI, and WinUI Gallery
- Q&A

### **Q&A Code of Conduct**
- To help ensure Q&A remains beneficial for the majority of the call's developer audience, please consider whether your question could be framed so that it might be more broadly relevant to our larger developer audience by ensuring it is reasonably concise, broadly beneficial, well-suited for the fast-paced live Q&A format, or able to spark open ended discussion.  
- Please understand some questions are not well-suited for the fast-paced live Q&A format and should be filed as standalone [Q&A](https://github.com/microsoft/microsoft-ui-xaml/discussions/new?category=q-a&title=%5BYour+Question+Here%5D) discussions where they can be most appropriately handled. Excluding presentation topic requests, this generally includes requests for dedicated follow-ups or technically complex questions that require investigative time or expertise not included in this month's guest list.

### Leave us your questions in the comments on this issue or live in the YouTube chat during the stream!

**Comment by bogdan-patraucean** [comment]
I have a few questions :)

1. Is it true that the UWP team will shift their efforts towards WinAppSDK? if yes, what would that mean for us?
2. What's with the absence of WinUI team in the UWP Community Discord channel? I think their presence there would encourage more developers to adopt this new tehnology.
3. What's the roadmap for this year? what should we expect?
4. How do you plan to promote WinAppSDK, should it be the next big thing in terms of desktop apps?

**Reply by llothar** [reply]
I want to highlight that point 4 is important. We can see from all measurable social media places that WinAppSDK is absolutely not taken serious right now.

**Reply by bogdan-patraucean** [reply]
point 4 was answered in the chat for this month's community [call](https://www.youtube.com/watch?v=y4gQY1hqp_I).

**Comment by gus33000** [comment]
Hello :)

I do have a few questions as well:

Considering the AppSDK is pivoting as the replacement for the legacy Windows.UI.Xaml platform:

- Is there any plan to include the missing controls inbox Windows.UI.Xaml had such as the MapsControl in the short to mid term? A few apps need these controls (specifically social apps)? Many of these apps cannot upgrade currently due to dependencies on these.
- Is there any plan to make the appsdk work on more devices (such as xbox or hololens)? If not, is there any technical limitation currently regarding these device families or is it just not planned?
- Is there any plan to ship the Windows App SDK inbox with Windows so users do not have to download a separate runtime to use applications?
- Are further updates expected this year, should we expect more previews/experimental channel releases? What roadmap are we expecting?
- Are there any plan to bring GameBar support with AppSDK or a compatibility to use older unmaintained nuget packages?

**Reply by JanRajnoha** [reply]
Question about availability on HoloLens should have in first step answered question about HoloLens future :/

**Reply by riverar** [reply]
There was a question here about whether or not the GameBar SDK worked with WinAppSdk. Adding the NuGet to a WinUI 3 project quickly answers that question (no).

**Reply by gus33000** [reply]
> There was a question here about whether or not the GameBar SDK worked with WinAppSdk. Adding the NuGet to a WinUI 3 project quickly answers that question (no).

I am aware that the sdk does not work, hence my question about if there are any plans to make it work :)

**Reply by riverar** [reply]
Right, sorry for the confusion. @DarranRowe deleted the question/comment I was responding to.

**Reply by DarranRowe** [reply]
@riverar Interestingly, I had deleted it before you even commented.

**Comment by riverar** [comment]
* What's the status of the WinUI repository cleanup effort?
* When is the roadmap going to get updated? (https://github.com/microsoft/microsoft-ui-xaml/blob/main/docs/roadmap.md)
* Is Product Board still being used? (It's in the README...) https://portal.productboard.com/winappsdk/1-windows-app-sdk/tabs/2-planned
* It was previously mentioned the WinUI team was heads down working primarily on partner tasks (e.g., Adobe, Apple.). Is that work done? Will the team be returning to work on community issues?

**Reply by JanRajnoha** [reply]
Productboard has been abandoned, because it has been maintained by member, who is not in Microsoft already. Better question is, if will be some alternative or reincarnation of this tool by team, not by member

**Reply by Panda-Sharp** [reply]
Regarding product board take a look here: https://github.com/microsoft/WindowsAppSDK/discussions/3356

**Reply by riverar** [reply]
@Panda-Sharp Not sure what I'm looking for there, but that doesn't explain why it's in the README.

**Reply by Panda-Sharp** [reply]
@riverar never mind

**Comment by mdtauk** [comment]
How long for WinUI open sourcing?

When will roadmaps and new features be detailed and set out?

WinUI designer progress?

Progress for more first party programs moving to WinUI 3 and WindowsAppSDK?

**Comment by saint4eva** [comment]
- When would you open source WinUI? 
- What is the status of cswinrt and cswin32? Are they still be developed or the effort is only on the c++ side?

**Reply by riverar** [reply]
Tagging @aarnott for heads up

**Reply by pjmlp** [reply]
Regarding cswinrt and cswin32, when can we expect the same tooling capabilities/developer productivity of .NET Native and C++/CX, without the manual work and low level interop code that these alternatives (+ c++/winrt) require?

**Reply by AArnott** [reply]
Yes, CsWin32 is still under development. It's mostly feature complete, and we are responding to bugs as well as filling in the rest of the feature gaps, as tracked by active issues.

AFAIK CsWin32 works with .NET Native, but I haven't tested this. We do have an active issue around eliminating all reliance on the .NET marshaler in the CsWin32-emitted code since I understand such reliance would fail in .NET Native.

**Reply by riverar** [reply]
Can vouch for @AArnott (if anyone needed that). Been working with him a lot on metadata side of things, which, to me, is a good indicator CsWin32 development is alive and well.

**Reply by AathifMahir** [reply]
> Yes, CsWin32 is still under development. It's mostly feature complete, and we are responding to bugs as well as filling in the rest of the feature gaps, as tracked by active issues.
> 
> AFAIK CsWin32 works with .NET Native, but I haven't tested this. We do have an active issue around eliminating all reliance on the .NET marshaler in the CsWin32-emitted code since I understand such reliance would fail in .NET Native.

That's awesome to hear and if it supports .net Native, Does it Also Support NativeAOT?

**Comment by JanRajnoha** [comment]
I have also few questions, targeted to touch:
- When we should see any progress on native support for touch keyboards? For my app, which will be used only by touch, it is extremely limiting -> #6291 
- I could not find any documentation for problem with touch keyboard showing over app, which is in Full screen mode. For maximized is used split screen - touch keyboard and rest of app. Is any way, how to show touch keyboard for app in full screen mode and still achieve automatic filling rest space by app? -> https://github.com/microsoft/WindowsAppSDK/discussions/3362
- Is any possible way to determine development of Windows App SDK? Had some tool, like mentioned above, ProductBoard or some alternative, where we could prioritize features
- Could you please share roadmap for next year, or at least 1-2 major releases? For what to expect from this framework

**Comment by hez2010** [comment]
Hello, I have some questions.

- When can we get NativeAOT support for WASDK & WinUI (I think the major work around it will happen in CsWinRT)?
- Given that UWP has been discontinued, what's the plan around Xbox so we can continuously deliver our apps to Xbox devices? Can we use WASDK for Xbox apps?
- The title bar coming with WinUI 3 has a lot of bugs and also it doesn't consist with the system title bar. Is there any plan to unify the WinUI 3 title bar and the WASDK title bar? (ref: https://github.com/microsoft/microsoft-ui-xaml/issues/7131)

**Reply by DarranRowe** [reply]
There was an issue on the Windows App SDK repo. You can find it [here](https://github.com/microsoft/WindowsAppSDK/issues/3346). It has an interesting reply that you may find useful.

**Comment by koenvd** [comment]
How can we get bugs prioritised on the WindowsAppSdk / WinUI3.0 backlog? Or get feedback when they're getting tackled?

**Comment by omikhailov** [comment]
Good morning! It would be interesting to talk not only about technical issues, but about the project as a whole too.

0. How it goes? 😊
1. How is your team structured? What does the development process look like? How do you make decisions? How do you choose people for the team?
2. Do you discuss ideas together or does one person come up with everything? What interesting ideas do you see now?
3. Most of the projects now are web projects. How do you rate Uno's experiment in using WinUI with Web Assembly, and can we expect Microsoft to also have a similar technology? Have you discussed this?
4. Do I understand correctly that right now you are mainly focused on the needs of various teams within Microsoft and on improving Windows 11?
5. Your partners outside of Microsoft, what kind of projects are they doing with the App SDK? Are they rewriting old apps or making new ones?

**Comment by horaciodiez** [comment]
Hi!
1) Is there any plan to modify ContentDialog to be movable? 
   I need to see what's underneath :)
2) What about this discussion? 
   [https://github.com/microsoft/microsoft-ui-xaml/discussions/7491](url)
Thank You!

**Reply by castorix** [reply]
> 1. Is there any plan to modify ContentDialog to be movable?
>    I need to see what's underneath :)

You can move it with _ManipulationMode_
I had posted a sample on MS forums :
https://learn.microsoft.com/en-us/answers/questions/847411/winui-real-dialogwindow

**Comment by MEK3DK** [comment]
What is the current status on SwapChainPanel is the team working on it? 

There is no WinUI 3 sample, the WinUI 3 docs reference a sample that has a VS2013 solution (does not even open in VS2022). Any plans to create a new WinUI SwapChainPanel sample?

There seems to be a series of bugs related to SwapChainPanel, which is the reason why I ask whether this control is being worked on in general. E.g.:
[WinUI application closed by 'X' but remains as dormant proc](https://github.com/microsoft/microsoft-ui-xaml/issues/7193)
[WinUI3: SwapChainPanel renders in wrong position when display scaling is not 100%](https://github.com/microsoft/microsoft-ui-xaml/issues/5888)
[Performance Issue, UI Thread ~30fps](https://github.com/microsoft/microsoft-ui-xaml/issues/7290)

**Reply by castorix** [reply]
> There is no WinUI 3 sample, the WinUI 3 docs reference a sample that has a VS2013 solution (does not even open in VS2022). Any plans to create a new WinUI SwapChainPanel sample?

In which language ?
I have many samples (like Direct2D basic animated sprites) using SwapChainPanel or Swap Chains without SwapChainPanel, but in C#, because it takes hours on my old PC to compile C++/WinRT...
I still have the resizing problem (with App SDK 1.1.0) for scale > 100%, that's why I often use Swap Chains without SwapChainPanel.

**Reply by MEK3DK** [reply]
I am not sure what you are asking @castorix?

If it is where the current WinUI C# sample can be found then it is here in the 1.2.0 docs: https://learn.microsoft.com/en-us/windows/windows-app-sdk/api/winrt/microsoft.ui.xaml.controls.swapchainpanel?view=windows-app-sdk-1.2 first line under "Initializing a SwapChainPanel element".

**Reply by smourier** [reply]
SwapChainPanel works fine for me, if you're looking for a C# sample of using WinUI3's SwapChainPanel, there's one here: https://github.com/smourier/DirectN/tree/master/DirectN/DirectN.WinUI3.MinimalD3D11 the SwapChainPanel's part is actually quite simple

**Comment by MEK3DK** [comment]
Any update on XamlIsland? Hosting WinUI 3 in WPF and hosting WPF in WinUI 3.

**Comment by Mysterious-Dev** [comment]
Any news about an equivalent solution of the [ApplicationView.SetPreferredMinSize(Size)](https://learn.microsoft.com/en-us/uwp/api/windows.ui.viewmanagement.applicationview.setpreferredminsize?view=winrt-22621) (UWP) method for the Windows App SDK? I know it's already possible (https://github.com/microsoft/microsoft-ui-xaml/issues/2945#issuecomment-776446712) but as 1.3 is "supposed" to simplify things for developers, it might be interesting.

**Reply by MEK3DK** [reply]
This bug related to the same issue has been closed as not planned: [Window's min size support](https://github.com/microsoft/WindowsAppSDK/issues/3163). The proposal mentioned in there is still open.

I know they said previously that they were looking at the Window classes so let's see.

It seems beoynd belief that you would not have properties like this. Is there any other UI framework where you cannot set e.g. MinWidth on a Window?

**Reply by Balkoth** [reply]
It was mentioned in [2797](https://github.com/microsoft/WindowsAppSDK/discussions/2797) by @rkarman that this might be queued up for the 1.3 planning cycle. Is this still true, if not this should be a top priority. Not including basic windowing stuff from the get got was already a big enough mistake.

**Reply by Scottj1s** [reply]
Here's an option: https://learn.microsoft.com/en-us/answers/questions/1062421/set-min-size-of-windows-with-windows-app-sdk

**Comment by MarkBrainfire** [comment]
https://www.winui3.org/news/winui-community-call-feb-15th-2023/
I'm relieved to see this Community Call  announced.  I look forward to joining WinUI Developers on the live stream!
![Feb 23 CC Announcement Fixed](https://user-images.githubusercontent.com/78389339/216480081-ba4ce66f-7c6a-4cb6-87f1-19cd6793e728.jpg)

**Reply by pjmlp** [reply]
Uhm, overlapping statement of joining forces with Uno, versus MAUI and Blazor, interesting.....

**Comment by MEK3DK** [comment]
What is the status on #6299?

**Comment by NVLudwig** [comment]
Any new components coming out? 
The latest Win UI Figma file seem to hint at some can you confirm that these will be developed?

**Comment by lhak** [comment]
Any update on https://github.com/microsoft/microsoft-ui-xaml/issues/3665? This issue basically blocks my application from using winui 3 (still having to use the now unsupported .net core 3 and XAML islands) and probably impacts all applications using win2d by leaking gigabytes of gpu memory.

**Comment by Dev-Taehui** [comment]
Please, share the development progress of XAML Islands for WinUI 3 (a.k.a. XAML Islands v2). XAML Islands v1 supports .NET Core 3.x which is end of life! I want to use the latest .NET runtime for my app. I think that it is time for releasing (or at least prereleasing) XAML Islands v2.

**Comment by Balkoth** [comment]
A couple of questions:

- When will theming, specifically [2943](https://github.com/microsoft/WindowsAppSDK/issues/2943) work out of the box?
- When will it no longer be possible to create applications with options that are not supported and crash the application on start [2597](https://github.com/microsoft/WindowsAppSDK/issues/2597)?
- Why are annoying bugs like [7609](https://github.com/microsoft/microsoft-ui-xaml/issues/7609) hanging around for so long?

**Comment by llothar** [comment]
Nobody is porting their 3rd party Win32/MFC controls at the moment (because they all dont see the WinUI3 future). 
So is it possible to do some Win32Island ? 

The window navigation and interaction and dialogs around the main window control (a complicated Graph and Text Control) would benefit from a modern look. But the hundert thousands of lines of the main window drawing is economically untouchable.
I never understood why the other way around with Xaml Island is so hyped. I know it's much easier to implement and looks like a solution for a problem for me. 

We need total interoperability to get to the future.

**Reply by pjmlp** [reply]
Why bother with porting when the aging MFC provides a much better Visual Studio experience than using C++/WinRT, after C++/CX was deprecated in 2016 and to this day, C++/WinRT still feels like using ATL from 2000?

C++ Builder and Qt/Qt Creator are the ones to look for a modem C++ experience, that C++/CX tried to provide, with the added benefit of being cross platform.

If we need to rewrite, at least for better development experience stacks.

**Reply by DarranRowe** [reply]
@llothar
Xaml Islands is easier to do compared to the other way around.
The Visual Layer, aka composition, has a well known quirk where composition drawn stuff always wins out over HWND stuff. This means that the Xaml host, which uses composition to draw things, would always have the Xaml content win out over any child window. So any HWND control host would have to communicate with the Xaml host to tell it to avoid drawing in certain areas.
This should be possible, but this is something that would definitely require planning and a feature proposal.

**Comment by anup-das** [comment]
When can we expect the ListView and GridView controls freezing issue to be fixed ?
Issue #8023 
Isssue #7972

**Comment by artkat** [comment]
@gabbbilka

I see you have planned another WinUI Community call.

I have asked several times to be considerate to people in east Asia (Taiwan, Japan, China). 
But again, you planned this meeting at 1:00 am for us. Some of the people in east Asia would like to participate in the live meeting and give live feedback.
If you don't value input from people on the other side of our planet, then just say so.

**Reply by riverar** [reply]
@artkat It's impossible to accomodate everyone on the planet, so please don't pretend there's a "good" time to pick. Also, these videos tend to get like 1K views or less, and feed mostly from the Q&A here, so hardly worth the effort in trying to accomodate everyone.

**Reply by shane-vorwerk** [reply]
I agree with this, I would be able to take part if it is earlier in the day rather that after midnight.  @gabbybilka  I suggest that each month you consider different time zones for community calls so everyone has the opportunity take part.

**Reply by artkat** [reply]
1. Don't call it a Global event, when you clearly just target the US (by only showing US time zones).
2. By starting the event 1 or 2 hours earlier, you will make it much easier for people in east Asia to join.
3. Why not organize the event at different times each time?

**Reply by MarkBrainfire** [reply]
My dear brother @artkat, your interest in WinUI technologies is VERY MUCH appreciated!

I understand your desire to have the times of the Community Call be more convenient for you.  I used to live in Thailand and in Bangkok there is a very active Microsoft development community.  In the future, I expect there will be WinUI Developer groups and conferences formed there.

In the meantime, I will be publishing the video of this Community Call here:

https://www.winui3.org/video-category/videos-winui-community-calls/

**Reply by gabbybilka** [reply]
Hi @artkat, @shane-vorwerk and others in East Asia and Oceanic time zones. Although we do our best to choose a time that accommodates the majority of the audience and our presenters, it's not possible to accommodate all. I wish we could! I could see us hosting a one-off session at a time zone that provides an opportunity for you to join us live, however I do not anticipate we will change the time we consistently host the sessions.

Additionally, as @MarkBrainfire mentioned we save the recordings of each live call for you to watch at a more convenient time for you: https://www.youtube.com/playlist?list=PLI_J2v67C23ZqsolUDaHoFkF1GKvGrttB

**Comment by AathifMahir** [comment]
Are there any plans for improvement to XAML in General, Xaml has always been the same from the beginning. Regardless of Modern Techniques and etc..

For Example:

- Custom Xaml Binding Like Maui
- Improvements for Compiled Bindings
Etc..

**Comment by nkasco** [comment]
I'm sure traditional app devs will give me the "don't do that" answer, but can the product team provide any information around a Hello World example that would load the proper types into PowerShell (5 or 7, either way)?

This is possible to do with WPF and WinForm, and many SysAdmins build tools this way. Would be AWESOME to be able to do it with WinUI.

I imagine a lot of it would be similar to WPF w/ XAML, but the knowledge in the community doesn't seem to be there to understand what types need to be loaded at runtime.

**Comment by Skittles2519** [comment]
Thanks for continued work on WinUi. My question is for the timeline of being able to print with WebView2 on Winui2.x Thank you

**Comment by eduardobragaxz** [comment]
I know you guys have said you're looking into a designer, but I'd like to add my voice to the choir. Hoping you guys bump it up as one of your priorities. Is there anything you can share about it at this moment?

**Reply by MarkBrainfire** [reply]
It was mentioned that the funds weren't available for a first class XAML designer.  I think this reflects a serious strategic mistake by Microsoft.  Desktops (and laptops) have ALWAYS been the domain of Microsoft. Today, 80% of all real computers (desktops, laptops) are running Microsoft Windows.  The power of these laptop computers has never been greater.  Almost ALL serious work is done on these machines.  Yet Microsoft's obsession with Azure and web apps that cater to mindless mobile phone scrollers is creating a world where every app depends upon the Internet and the "Cloud".  That's a fragile world -- particularly in today's geopolitical environment.  

Microsoft should pull out the stops to MAINTAIN DOMINANCE OF THE DESKTOP / LAPTOP market.  People should be able to run their software with or without an Internet connection.  WinUI is the key!  WinForms is STILL going strong today because it's so easy to use its drag and drop designer.  OF COURSE, we should have this kind of easy to use drag and drop designer for XAML.  It should be possible to quickly create WinUI 3 apps without ever needing to see a XAML page.  MVVM binding can be built into the designer.  RAD RAD RAD.  Remember?  Any of us could sit down in a few hours and knock out an impressive WinForms application..  It should be the same for WinUI 3 development.

And remember, WinUI 3 coupled with Uno Platform enables your WinUI 3 application to run LOCALLY on ANY desktop or computer (Mac OS and Linux) as well as all mobile devices.  (and WASM is an added bonus).  So WinUI 3 IS THE GATEWAY TO A NEW APPROACH FOR DEVELOPING SOFTWARE.

In the next few days you'll hear lots talk about Blazor Unified.  But does anyone want to develop a complex desktop application in Blazor (or Xamarin Forms for that matter).  NO!  We want to develop applications in C#, WinUI 3 and then use Uno Platform to enable those applications to run EVERYWHERE.   

So, coming up with the funds to develop a RAD XAML designer for WinUI 3 will enable Microsoft to maintain its desktop dominance.

**Comment by Diegorro98** [comment]
I would like to give some feedback and report some issues about the Widgets development, but I don't really know where I can post it. Maybe here is a good place? maybe on the issues of this repo (but because I didn't see any issue with this topic I wasn't sure?  or maybe there is an specific place for this, in that case, where it is?

Thanks!

**Comment by davepruitt** [comment]
What is the status of a XAML designer/previewer?

In this repository, the "WinUI 3.0 XAML Designer" discussion (https://github.com/microsoft/microsoft-ui-xaml/issues/5917) is 2nd most-commented issue and has the 3rd most likes of all issues.
In the .Net MAUI repository, the corresponding XAML designer issue (https://github.com/dotnet/maui/issues/3439) is the 4th most-commented issue and is 1st in the number of likes of all issues.

Clearly the customers want a XAML designer/previewer rather than Hot Reload. Can you please elaborate on the status of this feature?

**Comment by koenvd** [comment]
Can we get an update about https://github.com/MicrosoftEdge/WebView2Feedback/issues/3003 (aka _WinUI3 WebView2 mouse stop working while keyboard is still working_) since things became a bit silent after moving the issue to WindowsAppSDK team and it's still a high priority defect.

**Comment by jp-weber** [comment]
Why has the active further development of WinUi2 been discontinued, although WinUi3 is still in a disastrous state?

**Comment by terrycox** [comment]
I have a few questions:

1) I'd like to access a plug-in for a Windows Store app. I believe I can use code based on https://learn.microsoft.com/en-us/dotnet/core/tutorials/creating-app-with-plugin-support to load and access the plug-in DLL, but I'm at a loss as to how
to locate the library. Are there examples or documentation on how to do so for store apps?

2) Our app has many ComboBoxes, particularly editable ComboBoxes. We've experienced bind issues for these- not bind failures, but what look like failures to initiate the bind. Are you aware of any such issues?

**Comment by gabbybilka** [comment]
This live event has now ended, but you can view the recording on YouTube at the link above. If you didn't get your question answered, feel free to open a Q&A [discussion](https://github.com/microsoft/microsoft-ui-xaml/discussions/new?category=q-a&title=%5BYour+Question+Here%5D) or leave your question for the [next call](https://github.com/microsoft/microsoft-ui-xaml/discussions/8187). Thanks for joining!

**Reply by bogdan-patraucean** [reply]
Is there a reason why you didn't go through the questions asked here? besides feature related questions everything else was skipped. It's the first community call I participate but my expectations were that you will address all of them.

**Reply by saint4eva** [reply]
It has always been this way.

**Reply by saint4eva** [reply]
It has always been this way.

**Reply by riverar** [reply]
@bogdan-patraucean Here's what I pulled out of the video https://github.com/microsoft/microsoft-ui-xaml/discussions/8137#discussioncomment-5016881 I haven't gone through the chat stream, might be a few extra in there.

**Comment by MarkBrainfire** [comment]
It was mentioned that the funds weren't available for a first class XAML designer. I think this reflects a serious strategic mistake by Microsoft. Desktops (and laptops) have ALWAYS been the domain of Microsoft. Today, 80% of all real computers (desktops, laptops) are running Microsoft Windows. The power of these laptop computers has never been greater. Almost ALL serious work is done on these machines. Yet Microsoft's obsession with Azure and web apps that cater to mindless mobile phone scrollers is creating a world where every app depends upon the Internet and the "Cloud". That's a fragile world -- particularly in today's geopolitical environment.

Microsoft should pull out the stops to MAINTAIN DOMINANCE OF THE DESKTOP / LAPTOP market. People should be able to run their software with or without an Internet connection. WinUI is the key! WinForms is STILL going strong today because it's so easy to use its drag and drop designer. OF COURSE, we should have this kind of easy to use drag and drop designer for XAML. It should be possible to quickly create WinUI 3 apps without ever needing to see a XAML page. MVVM binding can be built into the designer. RAD RAD RAD. Remember? Any of us could sit down in a few hours and knock out an impressive WinForms application.. It should be the same for WinUI 3 development.

And remember, WinUI 3 coupled with Uno Platform enables your WinUI 3 application to run LOCALLY on ANY desktop or computer (Mac OS and Linux) as well as all mobile devices. (and WASM is an added bonus). So WinUI 3 IS THE GATEWAY TO A NEW APPROACH FOR DEVELOPING SOFTWARE.

In the next few days you'll hear lots of talk about "Blazor United". But does anyone want to develop a complex desktop application in Blazor (or Xamarin Forms for that matter)? No! We want to develop stateful, fast, applications in C#, WinUI 3 and then use Uno Platform to enable those applications to run EVERYWHERE.

So, coming up with the funds to develop a RAD XAML designer for WinUI 3 will enable Microsoft to maintain its desktop dominance.

**Comment by riverar** [comment]
Q&A transcribed from the video (not from the chat window).

@omikhailov
Q: How is your team structured? What does the development process look like? How do you make decisions? How do you choose people for the team? [...]
https://github.com/microsoft/microsoft-ui-xaml/discussions/8137#discussioncomment-4829000
A: [Long.] https://youtu.be/xcgc5CLqUkM?t=2272

@riverar
Q: Is Product Board still being used?
A: No. (https://youtu.be/xcgc5CLqUkM?t=2599)

@dotmorten
Q: [From chat] Can you set the backdrop in markup or do you have to continue using code-behind?
A: Yes [Fixed.] (https://youtu.be/xcgc5CLqUkM?t=2693)

@dotmorten
Q: Can you bind to the window backdrop [...]?
A: [Maybe. (Probably not.)] (https://youtu.be/xcgc5CLqUkM?t=2800)

Covers 4-5 general questions
Q: Is X feature coming?
A: Various answers. Maps, engaged with team. WinUI not happening on Xbox, HoloLens. etc. (https://youtu.be/xcgc5CLqUkM?t=2886)

Covers various questions
Q: Is a xaml designer coming?
A: Not focused on building one; is in the backlog. It is super costly to build. (https://youtu.be/xcgc5CLqUkM?t=3091)

Steve Terepin
Q: Is there a possibility of Win2D support in 1.4?
A: Win2D will remain standalone component https://youtu.be/xcgc5CLqUkM?t=3231

Covers various questions
Q: Module support in cppwinrt, faster builds in cppwinrt, etc.
A: cppwinrt effort is community effort, if you're interest in support for X, work with community (https://youtu.be/xcgc5CLqUkM?t=3333)
A: Aware of tooling gaps, is in the backlog, not sure when they'll get to it

Bogdan
Q: Performance improvements [...]
A: Did some perf tuning, will probably never be better (from .NET perspective) (https://youtu.be/xcgc5CLqUkM?t=3384)

Unknown
Q: Powershell with relation to loading WinAppSdk types into PowerShell
A: Not happening, bring your own solution (https://youtu.be/xcgc5CLqUkM?t=3471)

**Reply by saint4eva** [reply]
Wish WinUI would be a standalone component on top of WASDK

**Reply by riverar** [reply]
@Panda-Sharp Saw your emoji; is something about the transcript confusing?

**Reply by Panda-Sharp** [reply]
@riverar just disappointed by the replies

**Reply by riverar** [reply]
@Panda-Sharp Ah got it, thanks!
